dialects/postgresql/src/main/kotlin/app/cash/sqldelight/dialects/postgresql/grammar/mixins/LiteralValueMixin.kt

```diff
@@ -0,0 +1,36 @@
+package app.cash.sqldelight.dialects.postgresql.grammar.mixins
+
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlCreateTableStmt
+import com.alecstrong.sql.psi.core.psi.SqlInsertStmt
+import com.alecstrong.sql.psi.core.psi.SqlSetterExpression
+import com.alecstrong.sql.psi.core.psi.SqlTypes
+import com.alecstrong.sql.psi.core.psi.SqlValuesExpression
+import com.alecstrong.sql.psi.core.psi.impl.SqlLiteralValueImpl
+import com.intellij.lang.ASTNode
+import com.intellij.psi.tree.TokenSet
+import com.intellij.psi.util.PsiTreeUtil
+
+internal abstract class LiteralValueMixin(
+  node: ASTNode,
+) : SqlLiteralValueImpl(node) {
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    if (node.findChildByType(setterOnlyLiterals) != null) {
+      val values = PsiTreeUtil.getParentOfType(this, SqlValuesExpression::class.java)
+
+      if (values != null && values.parent is SqlInsertStmt) return
+      if (PsiTreeUtil.getParentOfType(this, SqlSetterExpression::class.java) != null) return
+      if (PsiTreeUtil.getParentOfType(this, SqlCreateTableStmt::class.java) != null) return
+
+      annotationHolder.createErrorAnnotation(this, "Cannot use time literal in expression")
```

Comment:
test case for this

Reply:
you can do error tests in the fixtures

Reply:
I mean specifically on this error annotation. They're super easy to set up a test for, it's like [this](https://github.com/AlecStrong/sql-psi/blob/master/test-fixtures/src/main/resources/fixtures/trigger-migration/1.s#L5-L6) where a test fixture has a comment specifically for a reported error

Reply:
Thanks - specifically I was interested in the inherited [LiteralValueMixin](https://github.com/AlecStrong/sql-psi/blob/master/core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/LiteralValueMixin.kt#L20-L27) that has the restriction `"Cannot use time literal in expression"`?

As both PostreSql + ansiFixtures are run by the tests - I can't override it in the PostgreSql dialect to support the syntax
`updatedAt > CURRENT_TIMESTAMP - INTERVAL '30' days` - The [timestamp-literal](  https://github.com/AlecStrong/sql-psi/blob/master/test-fixtures/src/main/resources/fixtures/timestamp-literals/Test.s#L25) will always run and currently fails if I override the restriction in LiteralValueMixin.

I looked at the Fixture setup and I don't think it will override the `ansi` test cases even if I redeclare the test fixture `timestamp-literals` in `postgreSql` fixtures.

Hopefully - that makes sense and is for another PR once I understand it :)